lib/sspi/sspwin32.cc

```diff
@@ -9,9 +9,23 @@
 #include "squid.h"
 #include "base64.h"
 #include "ntlmauth/ntlmauth.h"
-#include "sspwin32.h"
+#include "sspi/sspwin32.h"
 #include "util.h"
 
+#if HAVE_WINDOWS_H && HAVE_SSPI_H
+
+// FARPROC is an exception on Windows to the -Wcast-function-type sanity check.
+// suppress the warning only when casting FARPROC
+template<typename T>
+T
+farproc_cast(FARPROC in)
+{
+#pragma GCC diagnostic push
```

Comment:
```
warning C4068: unknown pragma 'GCC'
```

To protect us from compiler warnings similar to the above, please wrap these pragmas in #ifdefs. There are examples in the official code.

AFAICT, we can assume that modern clang and GCC versions support these pragmas but compilers like MSVC [do not](https://godbolt.org/z/W51oT6rKn). Adding a `__GNUC__` wrapper helps in my tests, but there may be a better way.

Reply:
Where did you get that warning from?

The ```GCC``` prefix is supposed to be standard way of defining the rest of the line as GCC-only and to be ignored by all other C++ compilers.

Reply:
```
warning C4068: unknown pragma 'GCC'
```
 
> > Alex: AFAICT, we can assume that modern clang and GCC versions support these pragmas but compilers like MSVC [do not](https://godbolt.org/z/W51oT6rKn).

> Amos: Where did you get that warning from?

The problem is general, but that particular warning came from an MSVC test that was linked from the original change request: https://godbolt.org/z/W51oT6rKn

> The GCC prefix is supposed to be standard way of defining the rest of the line as GCC-only and to be ignored by all other C++ compilers.

It _is_ ignored, with a warning.

As for GCC prefix being "standard way", I see no such language/mechanism in pragma [documentation](https://en.cppreference.com/w/cpp/preprocessor/impl). There is a requirement to _ignore_ pragmas that are not recognized. AFAICT, that requirement does not prohibit issuing warnings (or supporting GCC pragmas in a non-GCC compiler for that matter).

BTW, these problems with pragmas is one of the many examples illustrating why I consider defaulting to `-Werror` a mistake, but this change request is not about that.

Reply:
Fixed in [9a918aa](https://github.com/squid-cache/squid/pull/1348/commits/9a918aaaacb1a9d5c0a7c834a591c7cfaf5a6005)